src/fsharp/fsi/fsi.fs

```diff
@@ -821,7 +821,8 @@ type internal FsiDynamicCompiler
         errorLogger.AbortOnError();
             
         let fragName = textOfLid prefixPath 
-        let codegenResults = GenerateIlxCode (IlReflectBackend, isInteractiveItExpr, runningOnMono, tcConfig, topCustomAttrs, optimizedImpls, fragName, true, ilxGenerator)
+        let providedTypes, _staticLinker = Driver.StaticLinker.StaticLink(tcConfig, tcImports, ilGlobals)
```

Comment:
Hmmm... I see "_staticLinker" is ignored here.  I'm confused. I'm guessing that's because all the actual static linking is now done in IlxGen.fs.  

Now I'm wondering why the process of folding the provided types with the codegenResults isn't done in the StaticLink code? Doing it in IlxGen.fs seems somehow wrong - couldn't we have adjusted adjust implantTypeDef (which has been removed? because it's now done In IlxGen.fs?) to merge the type definitions where necessary?

I guess I now don't understand why fsc.exe uses one static linking code, and fsi.exe uses another.